src/SFA.DAS.EmploymentCheck.Data/Repositories/EmploymentCheckRepository.cs

```diff
@@ -145,10 +145,11 @@ public async Task UpdateEmploymentCheckAsComplete(Models.EmploymentCheckCacheReq
             parameter.Add("@apprenticeEmploymentCheckId", request.ApprenticeEmploymentCheckId, DbType.Int64);
             parameter.Add("@employed", request.Employed, DbType.Boolean);
             parameter.Add("@requestCompletionStatus", (short)ProcessingCompletionStatus.Completed, DbType.Int16);
+            parameter.Add("@errorType", request.Employed == null ? "HmrcFailure" : null, DbType.String);
```

Comment:
this logic should not be here, make it an auto-property on the EmploymentCheck model? i.e. `string Failure => Employed == null ? "HmrcFailure" : null` don't forget to add unit tests ;)

Reply:
The existing code used the EmploymentCheckCacheRequest model to update the EmploymentCheck repository. Refactored to use an EmploymentCheck model to update the EmploymentCheck repository and moved the logic to the method that maps from the CacheRequest model the the EmploymentCheck model.